jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java

```diff
@@ -158,6 +159,8 @@ public void addCustomizer(Customizer customizer)
 
     public List<Customizer> getCustomizers()
     {
+        if (_customizers.isEmpty())
+            return Collections.emptyList();
```

Comment:
@sbordet  does this ensure faster iteration or should I not bother?

Reply:
I would not bother, also because before the `List` we returned was mutable, so people could do:

```java
httpConfig.getCustomizers().add(new SecureRequestCustomizer());
```

If you return `Collections.emptyList()` the `List` will be immutable.

Reply:
That method should have returned an iterable instead.
We have many places in the codebase where we cannot modify the collection returned, as the internals of the class returning the collection need to perform actions on add/remove of entries.